### PR TITLE
Jeg 19

### PR DESCRIPTION
**Pull request by Mikul17**

### CODE REVIEW

File: components/settings/CustomInputField.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        "& .MuiOutlinedInput-root": {
+          "& > fieldset": {
+            borderColor: props.borderColor ? props.borderColor : alpha(colors.primary[500], 0.4),
+            borderWidth: "0.15rem",
+            borderRadius: "0.75rem",
+          },
+          "&:hover fieldset": { 
+            borderColor: props.borderColor ? props.borderColor : colors.primary[500],
+            transition:"0.35s ease-in-out"
+        },
```
Comment (majer565) [inline components/settings/CustomInputField.tsx]: When input on focus color of the moved placeholder must be black or white depending on theme now its primary

File: __tests__/ChangePasswordModal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import SettingsModal from "@/components/settings/SettingsModal";
```
Comment (majer565) [inline __tests__/ChangePasswordModal.test.tsx]: Something is not okay with that stepper. Take a closer look when you change from step first to sec this stepper container moves some pixels which cause second step circle being cut (on safari)

File: components/settings/CustomInputField.tsx (hunk 1 above)
Comment (Mikul17) [reply components/settings/CustomInputField.tsx]: fixed

File: __tests__/ChangePasswordModal.test.tsx (hunk 2 above)
Comment (Mikul17) [reply __tests__/ChangePasswordModal.test.tsx]: fixed

File: components/settings/changePasswordStepper/PswrdChangeStepSecond.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { Box, Button, Typography, alpha, useTheme } from "@mui/material";
```
Comment (majer565) [inline components/settings/changePasswordStepper/PswrdChangeStepSecond.tsx:1]: Could you make dummy POST request which will respond with Promise object with a operation status? like status: 'failed' | 'success' | 'wrong-code'? Then if the code was wrong "NASTĘPNY" button will be blocked and maybe some error color text will be displayed under "WYŚLIJ PONOWNIE" e.g. "Podany kod jest błędny". If its failure (means something wrong with backend) button "NASTĘPNY" won't be disabled but final step will be as error in figma. And last success as it is now

**Review by majer565 — CHANGES_REQUESTED** [review]

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+
+interface ChangePasswordModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+const ChangePasswordModal = (props: ChangePasswordModalProps) => {
+  const theme = useTheme();
+  const colors = tokens(theme.palette.mode);
+  const [activeStep, setActiveStep] = React.useState(0);
```
Comment (majer565) [inline components/settings/changePasswordStepper/ChangePasswordModal.tsx]: Always add type to useState -> `React.useState<type>`

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+    },
+    "& .QontoStepIcon-circle": {
+      width: 8,
+      height: 8,
+      borderRadius: "50%",
+      backgroundColor: "currentColor",
+    },
+  }));
+
+  function StyledStepIcon(props: StepIconProps) {
```
Comment (majer565) [inline components/settings/changePasswordStepper/ChangePasswordModal.tsx]: use arrow function

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+            case 0:
+              return <PswrdChangeStepFirst />;
+            case 1:
+              return <PswrdChangeStepSecond />;
+            case 2:
+              return <PswrdChangeStepThird />;
+            default:
+              return <Typography>Coś poszło nie tak</Typography>;
+          }
+        })()}
```
Comment (majer565) [inline components/settings/changePasswordStepper/ChangePasswordModal.tsx]: extract switch to some function + remove this () at line 191

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx [hunk 7]
```diff
@@ -0,0 +1,269 @@
...
+            <NavigateBeforeRounded />
+            <Typography variant="h4">Poprzedni</Typography>
+          </Button>
+          <Button
+            onClick={
+              activeStep === 2
+                ? () => {
+                    props.onClose();
+                    handleReset();
+                  }
```
Comment (majer565) [inline components/settings/changePasswordStepper/ChangePasswordModal.tsx]: extract it to some function

**Review by majer565 — CHANGES_REQUESTED** [review]

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx (hunk 4 above)
Comment (Mikul17) [reply components/settings/changePasswordStepper/ChangePasswordModal.tsx]: fixed

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx (hunk 5 above)
Comment (Mikul17) [reply components/settings/changePasswordStepper/ChangePasswordModal.tsx]: fixed

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx (hunk 6 above)
Comment (Mikul17) [reply components/settings/changePasswordStepper/ChangePasswordModal.tsx]: fixed

File: components/settings/changePasswordStepper/ChangePasswordModal.tsx (hunk 7 above)
Comment (Mikul17) [reply components/settings/changePasswordStepper/ChangePasswordModal.tsx]: fixed

File: components/settings/changePasswordStepper/PswrdChangeStepSecond.tsx (hunk 3 above)
Comment (Mikul17) [reply components/settings/changePasswordStepper/PswrdChangeStepSecond.tsx:1]: fixed

**Pull request merged**